### PR TITLE
[Android] Brave news - cards scroll fixes and tweaks

### PR DESCRIPTION
**Pull request by alexsafe**
Resolves https://github.com/brave/brave-browser/issues/20773
Resolves https://github.com/brave/brave-browser/issues/20772
Resolves https://github.com/brave/brave-browser/issues/20410
Resolves https://github.com/brave/brave-browser/issues/20769
Resolves https://github.com/brave/brave-browser/issues/20771
Resolves https://github.com/brave/brave-browser/issues/20769
Resolves https://github.com/brave/brave-browser/issues/20768
Resolves https://github.com/brave/brave-browser/issues/20766

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java [hunk 1]
```diff
@@ -588,24 +597,52 @@ protected void onAttachedToWindow() {
 
         if (ChromeFeatureList.isEnabled(BraveFeatureList.BRAVE_NEWS)) {
             if (mSettingsBar != null) {
-                mSettingsBar.setVisibility(View.VISIBLE);
                 mSettingsBar.setAlpha(0f);
+                mSettingsBar.setVisibility(View.INVISIBLE);
             }
             InitBraveNewsController();
+            initNews();
+            Tab tab1 = BraveActivity.getBraveActivity().getActivityTab();
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java:605]: A check `if (BraveActivity.getBraveActivity() != null` should probably be before that line

File: android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java [hunk 2]
```diff
@@ -644,11 +681,18 @@ protected void onDetachedFromWindow() {
                 BraveActivity.getBraveActivity().setNewsItemsFeedCards(mNewsItemsFeedCard);
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java:681]: there are no checks on `if (BraveActivity.getBraveActivity() != null) {` but I see them in other places. Should we check it here as well?

File: android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java [hunk 3]
```diff
@@ -644,11 +681,18 @@ protected void onDetachedFromWindow() {
...
             }
 
-            if (mSettingsBar != null && mCompositorView != null) {
-                if (mSettingsBar.getId() == R.id.news_settings_bar) {
-                    mSettingsBar.setVisibility(View.GONE);
-                    mSettingsBar.setAlpha(0f);
-                }
+            Tab tab = BraveActivity.getBraveActivity().getActivityTab();
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java:684]: there are no checks on `if (BraveActivity.getBraveActivity() != null) {` but I see them in other places. Should we check it here as well?

File: android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java (hunk 1 above)
Comment (alexsafe) [reply android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java]: @SergeyZhukovsky the whole "tab1" block was useless there i think it was part of some dev logs . Same as below

File: android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java (hunk 3 above)
Comment (alexsafe) [reply android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java]: @SergeyZhukovsky i removed the entire "Tab" block. leftover

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Comment by srirambv** [comment]
Verification passed on Oppo Reno 5 with Android 12 and/or Samsung Tab A with Android 10 running 1.37.55 x64 nightly build

<details>
<summary><a href=https://github.com/brave/brave-browser/issues/20773>brave/brave-browser#20773</a> - <b>Passed</b></summary>
<br>
 
- Verified news feed doesn't repeat when scrolled down 
- Verified disabling and re-enabling Brave news doesn't cause the issue again

https://user-images.githubusercontent.com/17010094/154445053-c980b97e-f3f8-41c4-978c-d588c091be84.mp4

</details>

<details>
<summary><a href=https://github.com/brave/brave-browser/issues/20772>brave/brave-browser#20772</a> - <b>Passed</b></summary>
<br>

- Verified disable and re-enable Brave News, start to show the feed immediately within the same NTP

https://user-images.githubusercontent.com/17010094/154447093-003a4105-699e-4f4c-99e8-745b20b94b7a.mp4


</details>

<details>
<summary><a href=https://github.com/brave/brave-browser/issues/20410>brave/brave-browser#20410</a> - <b>Passed</b></summary>
<br>

- Verified clicking on the widget menu loads the menu options and able to go into widget settings
- Verified tapping on stats widget brings up the stats details page
- Verified tapping on Binance button loads the Binance widget 
- Verified tapping on a top tile loads the page 

https://user-images.githubusercontent.com/17010094/154450660-c15b694f-ad0f-48c7-a175-a4afcbfce787.mp4



</details>

<details>
<summary><a href=https://github.com/brave/brave-browser/issues/20771>brave/brave-browser#20771</a> - <b>Failed</b></summary>
<br>

- Verification failed for this one as it still repro's. Logged [ brave-browser#21164](https://github.com/brave/brave-browser/issues/21164)
 </details>

<details>
<summary><a href=https://github.com/brave/brave-browser/issues/20769>brave/brave-browser#20769</a> - <b>Passed</b></summary>
<br>

- Verified opening and closing new tabs via tab tray doesn't cause any crash
- Verified for both tab group enabled and disbaled
- Logged followup issue [#21167](https://github.com/brave/brave-browser/issues/21167)

https://user-images.githubusercontent.com/17010094/154453124-e66b1ded-d82f-4718-9a7b-0cf12fae04a2.mp4


</details>
<details>
<summary><a href=https://github.com/brave/brave-browser/issues/20768>brave/brave-browser#20768</a> - <b>Passed</b></summary>
<br>

- Verified brave news settings bar doesn't show up on tab tray when tabs are closed 

https://user-images.githubusercontent.com/17010094/154455080-1d7ac25a-ff98-443e-86d6-e36347c67953.mp4


</details>

<details>
<summary><a href=https://github.com/brave/brave-browser/issues/20766>brave/brave-browser#20766</a> - <b>Partial Pass or Failed</b></summary>
<br>

- Verified steps from issue description. But found the news bar is still shown on tab tray and the UI transition seems pretty bad. 
- Logged followup issue [#21169](https://github.com/brave/brave-browser/issues/21169)
</details>

* Encountered [#21166](https://github.com/brave/brave-browser/issues/21166)

**Pull request merged**